Fixed generate button on pathway analysis execution page

vlbe opened the pull request:
Fixes #2069

vlbe commented:
@anthonysena I agree. I'll prepare new commit.

anthonysena commented:
Excellent - thanks for that. I suspect that Estimation/Prediction also have this generation experience.

vlbe commented:
@anthonysena Hello! I've extracted execution tab content to separate component which will be shared between Cohort Pathway/Characterization/PLE/PLP and removed bunch of files that are not needed anymore. I've tested this on my local environment but i'll be very appreciated if you test it too.

There are some differences between Cohort Pathway/Characterization and PLE/PLP:
- Cohort Pathway/Characterization  execution tab - has an ability to cancel generation + has a  Design column + has link to view reports in Results column
- PLE/PLP execution tab - doesn't have an ability to cancel generation but you can start another one in parallel + doesn't have a Design column + has link to download results in zip instead of view reports in Results column.
- Some differences between conditions to execute, so some `extraExecutionPermissions` can be passed to the componentParams

And i have a question how will be better to implement error handing when loading of design failed. For now, i added alert to notify users that loading of design failed ( https://github.com/OHDSI/Atlas/pull/2118/files#diff-77e919ff16fbf32d1d863df621f0eb70R178 ). Maybe it will better to implement one of this:
1)  Open modal window with loading status(as now) and if error occurred - show error in this modal.
2) Do not open modal window before request will be fulfilled. Instead of this add some loading indicator to table cell and when request will be fulfilled - show modal with design/show alert error.
3) Leave as it is now.

What do you think about this?

anthonysena commented:
> And i have a question how will be better to implement error handing when loading of design failed. For now, i added alert to notify users that loading of design failed ( https://github.com/OHDSI/Atlas/pull/2118/files#diff-77e919ff16fbf32d1d863df621f0eb70R178 ). Maybe it will better to implement one of this:
> 
> 1. Open modal window with loading status(as now) and if error occurred - show error in this modal.
> 2. Do not open modal window before request will be fulfilled. Instead of this add some loading indicator to table cell and when request will be fulfilled - show modal with design/show alert error.
> 3. Leave as it is now.
> What do you think about this?

I think that using the alert mechanism as you have done in this PR is fine for now.

anthonysena commented on js/pages/characterizations/components/characterizations/characterization-view-edit/characterization-exec-wrapper.js:
```diff
@@ -3,20 +3,37 @@ define([
+                runExecutionInParallel: false,
+                resultsPathPrefix: '/cc/characterizations/',
+                analysisId: params.characterizationId,
+                ExecutionService: CharacterizationService,
```
The `CharacterizationService` is lacking a 'generate' function which then causes a failure when attempting to generate a characterization.
